experimental/native_file_system/virtual_root_provider_win.cc

```diff
@@ -1,12 +1,63 @@
-// Copyright (c) 2014 Intel Corporation. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-#include "xwalk/experimental/native_file_system/virtual_root_provider.h"
-
-#include "base/logging.h"
-
-VirtualRootProvider::VirtualRootProvider() {
-  // TODO(darktears): Windows support to be added.
-  NOTIMPLEMENTED();
-}
+// Copyright (c) 2015 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "xwalk/experimental/native_file_system/virtual_root_provider.h"
+
+#include <shlobj.h>
+
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+
+bool VirtualRootProvider::testing_enabled_ = false;
+
+VirtualRootProvider::VirtualRootProvider() {
+  if (!virtual_root_map_.empty())
+    return;
+
+  if (testing_enabled_)
+    base::GetTempDir(&home_path_);
+  else
+    home_path_ = base::GetHomeDir();
+
+  LPWSTR wszPath = NULL;
+  HRESULT hr = SHGetKnownFolderPath(FOLDERID_Desktop,
+                                    KF_FLAG_DONT_UNEXPAND, NULL, &wszPath);
+  if (SUCCEEDED(hr))
+    virtual_root_map_["DESKTOP"] = base::FilePath(wszPath);
```

Comment:
Free `wszPath` by `CoTaskMemFree()`.
That's a Windows API pattern, the calling process is responsible for freeing this resource once it is no longer needed by calling CoTaskMemFree(), refer to https://msdn.microsoft.com/en-us/library/windows/desktop/bb762188(v=vs.85).aspx